### PR TITLE
Moved to consolidated hub server

### PR DESCRIPTION
**Pull request by santanusinha**
This concludes and subsumes the work started in [https://github.com/appform-io/ranger/pull/24](https://github.com/appform-io/ranger/pull/24)

### CODE REVIEW

File: ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/configuration/RangerHttpUpstreamConfiguration.java [hunk 1]
```diff
@@ -13,30 +13,36 @@
...
+@ToString(callSuper = true)
 @JsonIgnoreProperties(ignoreUnknown = true)
-public class HttpAppConfiguration extends Configuration {
-    @NotEmpty
-    @NotNull
-    String name;
-    @Valid
-    @NotNull
-    RangerHttpConfiguration rangerConfiguration;
+public class RangerHttpUpstreamConfiguration extends RangerUpstreamConfiguration {
```
Comment (koushikr) [inline ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/configuration/RangerHttpUpstreamConfiguration.java:34]: Can use SuperBuilder and get away with the constructor.

File: ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/configuration/RangerServerConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package io.appform.ranger.hub.server.bundle.configuration;
```
Comment (koushikr) [inline ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/configuration/RangerServerConfiguration.java:1]: License missing

File: ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/RangerHubServerBundle.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+                .map(curatorFramework -> (HealthCheck) new RangerHealthCheck(curatorFramework))
+                .collect(Collectors.toList());
+    }
+
+    private class HubCreatorVisitor
+            implements RangerConfigurationVisitor<List<RangerHubClient<ShardInfo,
+            ListBasedServiceRegistry<ShardInfo>>>> {
+        private final String namespace;
+
+        public HubCreatorVisitor(String namespace) {
```
Comment (koushikr) [inline ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/RangerHubServerBundle.java:86]: Can use an AllArgsConstructor

File: ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/configuration/RangerConfigurationVisitor.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package io.appform.ranger.hub.server.bundle.configuration;
+
```
Comment (koushikr) [inline ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/configuration/RangerConfigurationVisitor.java:2]: License missing

File: ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/configuration/RangerUpstreamConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import javax.validation.constraints.NotNull;
+
+@AllArgsConstructor
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXISTING_PROPERTY, property = "type", visible = true)
+@JsonSubTypes({
+    @JsonSubTypes.Type(value = RangerHttpUpstreamConfiguration.class, name = "HTTP"),
+    @JsonSubTypes.Type(value = RangerZkUpstreamConfiguration.class, name = "ZK")
+})
+@Getter
+public abstract class RangerUpstreamConfiguration {
```
Comment (koushikr) [inline ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/configuration/RangerUpstreamConfiguration.java:35]: Like the above, Can use SuperBuilder and remove the AllArgsConstructor, for brevity

File: ranger-server/src/main/java/io/appform/ranger/server/App.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
+package io.appform.ranger.server;
```
Comment (koushikr) [inline ranger-server/src/main/java/io/appform/ranger/server/App.java:1]: License missing

File: ranger-server/src/main/java/io/appform/ranger/server/AppConfig.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
+package io.appform.ranger.server;
```
Comment (koushikr) [inline ranger-server/src/main/java/io/appform/ranger/server/AppConfig.java:1]: Same. License.

File: ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/configuration/RangerHttpUpstreamConfiguration.java (hunk 1 above)
Comment (santanusinha) [reply ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/configuration/RangerHttpUpstreamConfiguration.java:34]: we need to send type super ...

File: ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/configuration/RangerUpstreamConfiguration.java (hunk 5 above)
Comment (santanusinha) [reply ranger-hub-server-bundle/src/main/java/io/appform/ranger/hub/server/bundle/configuration/RangerUpstreamConfiguration.java:35]: there is no base class for this. The type parameter is mandatory to be sent by subclass. Constructor would be needed.

**Review by koushikr — APPROVED** [review]
Lgtm

**Review by koushikr — APPROVED** [review]

**Pull request merged**